iamy/aws.go

```diff
@@ -176,6 +190,27 @@ func (a *AwsFetcher) marshalRoleDescriptionAsync(roleName string, target *string
 	}()
 }
 
+func (a *AwsFetcher) populateInstanceProfileData(resp *iam.ListInstanceProfilesOutput) error {
+	for _, profileResp := range resp.InstanceProfiles {
+		if cfnResourceRegexp.MatchString(*profileResp.InstanceProfileName) {
+			log.Printf("Skipping CloudFormation generated instance profile %s", *profileResp.InstanceProfileName)
+			continue
+		}
+		profile := InstanceProfile{iamService: iamService{
+			Name: *profileResp.InstanceProfileName,
+			Path: *profileResp.Path,
+		}}
+		if len(profileResp.Roles) > 0 {
```

Comment:
you don't need to wrap the `for` loop in a length check since golang handles when the range is empty. see https://play.golang.org/p/UUhRUwypIk for an example.

Reply:
Ahh, yes.  I've cleaned it  up in 728fc67 .  That was a hail mary debugging attempt when I had:
```
type InstanceProfile struct {
        iamService              `json:"-"`
        Roles                   []string        `json:"Roles",omitempty`
}
```
instead of (note the quotes in the omitempty bit)
```
type InstanceProfile struct {
        iamService              `json:"-"`
        Roles                   []string        `json:"Roles,omitempty"`
}
```
and It was emitting `Roles: null` into the YAML instead of ignoring an empty Roles attribute.